server/src/main/java/org/tctalent/server/api/admin/SystemAdminApi.java

```diff
@@ -380,6 +385,29 @@ public DuolingoVerifyScoreResponse verifyScore(
         return duolingoApiService.verifyScore(certificateId, birthdate);
     }
 
+    /**
+     * Reassigns a new Duolingo coupon to a candidate and marks any previously assigned coupons as redeemed.
+     *
+     * @param candidateId The ID of the candidate to whom the new coupon will be assigned.
+     * @return A ResponseEntity containing the details of the newly assigned coupon.
+     * @throws NoSuchObjectException If the candidate or available coupons are not found.
+     */
+    @GetMapping("reassign-duolingo-coupon/{candidateId}")
```

Comment:
```suggestion
    @GetMapping("reassign-duolingo-coupon/{candidateNumber}")
```

Candidate number is better (which is a String, not a long). That is what is displayed everywhere on the TC browser app. Candidate id is only used internally. Candidate id is not the same as Candidate number. We have both for legacy reasons. One day there is a tech debt issue to get rid of Candidate ids and make Candidate number the primary key (and convert it to a long).

Reply:
Thanks for letting me know!